app/plugin_api.go

```diff
@@ -57,6 +63,39 @@ func (api *PluginAPI) LoadPluginConfiguration(dest interface{}) error {
 	}
 }
 
+func (api *PluginAPI) ValidateConfig(config map[string]interface{}) bool {
+	// walk api down to manifest to see if settings meta data is defined
+	// plugin api infratructure might not be completly initialized
+	if config == nil || len(config) < 1 ||
+		api == nil ||
+		api.manifest == nil ||
+		api.manifest.SettingsSchema == nil ||
+		api.manifest.SettingsSchema.Settings == nil {
+		return false
+	}
+	ss := api.manifest.SettingsSchema.Settings
+	var result = true // assume we succeed
+	for _, s := range ss {
+		required := s.Required
+		if !required { // if required is undefined or false -> skip check for this key
+			// TODO considering loggin a warning if required is nil (undefined)
```

Comment:
2/5 `required` is not required :) - remove the comment?

Reply:
will do ...